<p style="color:#0000FF";><b>
* 자료는 보통 순서대로 실행한다는 가정 하에 만듭니다. 제대로 실행이 되지 않는다면 이전의 셀을 실행했는지 다시 확인해 보세요. <br>
* ctrl+enter는 선택한 셀의 코드를 실행하고, shift+enter는 해당 셀 실행 후 다음 셀로 넘어갑니다.
    </b></p>

#### 0. 복습

#### 1. 데이터 파일 읽고 파악하기

#### 2. 한국미디어패널의 인구통계학적 정보 살펴보기


# 0. 복습

- 함수는 프로그램에서 한 번 이상 실행할 수 있도록 여러 코드를 그룹화하는 장치이며, 입력(input)과 출력(output)을 받아 연산을 수행할 수 있습니다.

- 파이썬은 여러 내장 함수를 제공하며, help(함수명)으로 설명을 조회할 수 있습니다.

- 메서드(method)는 함수의 한 형태로, 특정 자료형에서만 동작하는 함수를 말합니다.

<img src='https://github.com/eunyu1031/Media_interface_2020Spr/blob/master/pics/inout.png?raw=true'/>

In [1]:
def CalNums(x, y):
    result = x - y
    return result

mynum = CalNums(10, 7)
print(mynum)

3


In [2]:
help(print)

Help on built-in function print in module builtins:

print(...)
    print(value, ..., sep=' ', end='\n', file=sys.stdout, flush=False)
    
    Prints the values to a stream, or to sys.stdout by default.
    Optional keyword arguments:
    file:  a file-like object (stream); defaults to the current sys.stdout.
    sep:   string inserted between values, default a space.
    end:   string appended after the last value, default a newline.
    flush: whether to forcibly flush the stream.



In [3]:
print(82, 10, 1234, 5678, sep = '-')

82-10-1234-5678


In [4]:
help(str.split)

Help on method_descriptor:

split(self, /, sep=None, maxsplit=-1)
    Return a list of the words in the string, using sep as the delimiter string.
    
    sep
      The delimiter according which to split the string.
      None (the default value) means split according to any whitespace,
      and discard empty strings from the result.
    maxsplit
      Maximum number of splits to do.
      -1 (the default value) means no limit.



In [5]:
test2 = "2020-03-30"

result2 = test2.split('-')
print(result2, type(result2))

['2020', '03', '30'] <class 'list'>


DataFrame에서의 특정 값을 확인하는 방법들도 살펴보았습니다.
- DataFrame.**iloc[]**은 색인의 __순서를 나타내는 정수값(index)__을 받습니다.
- DataFrame.**loc[]**은 색인의 __라벨(label)__을 받습니다.
- Boolean indexing은 조건을 만족하는(True) 행을 가져올 수 있습니다.

<img src='https://github.com/eunyu1031/Media_interface_2020Spr/blob/master/pics/matrix.png?raw=true' width='400' height='400' >

Pandas: Indexing and selecting data: 
https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html

pandas 모듈을 가져올 때 Series, DataFrame은 바로 가져오고, 그 외의 요소들은 pd로 받아왔기 때문에  DataFrame은 pd.DataFrame()이 아닌 DataFrame()으로 실행하고 버전은 pd.~으로 확인합니다.

In [6]:
from pandas import Series, DataFrame 
import pandas as pd

pd.__version__ 

'1.0.5'

In [7]:
holiday_dict = {'name':['삼일절', '어린이날', '부처님 오신 날', 
                        '광복절', '한글날', '기독탄신일'],
                'start_year':[1949, 1975, 1975, 1949, 1949, 1949],
                'moon':[False, False, True, False, False, False]}

holidays = DataFrame(holiday_dict, 
                     index = ["3월", "5월", "음력4월", "8월","10월", "12월"])
holidays

name  start_year   moon
3월         삼일절        1949  False
5월        어린이날        1975  False
음력4월  부처님 오신 날        1975   True
8월         광복절        1949  False
10월        한글날        1949  False
12월      기독탄신일        1949  False

DataFrame의 요소(attribute)를 살펴볼 수 있었고, 메서드(method) 중 sort_values를 활용해 보았습니다.

__Attributes__
<table class="longtable table">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="pandas.DataFrame.T.html#pandas.DataFrame.T" title="pandas.DataFrame.T"><code class="xref py py-obj docutils literal notranslate"><span class="pre">T</span></code></a></p></td>
<td><p>Transpose index and columns.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="pandas.DataFrame.at.html#pandas.DataFrame.at" title="pandas.DataFrame.at"><code class="xref py py-obj docutils literal notranslate"><span class="pre">at</span></code></a></p></td>
<td><p>Access a single value for a row/column label pair.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="pandas.DataFrame.attrs.html#pandas.DataFrame.attrs" title="pandas.DataFrame.attrs"><code class="xref py py-obj docutils literal notranslate"><span class="pre">attrs</span></code></a></p></td>
<td><p>Dictionary of global attributes on this object.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="pandas.DataFrame.axes.html#pandas.DataFrame.axes" title="pandas.DataFrame.axes"><code class="xref py py-obj docutils literal notranslate"><span class="pre">axes</span></code></a></p></td>
<td><p>Return a list representing the axes of the DataFrame.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="pandas.DataFrame.columns.html#pandas.DataFrame.columns" title="pandas.DataFrame.columns"><code class="xref py py-obj docutils literal notranslate"><span class="pre">columns</span></code></a></p></td>
<td><p>The column labels of the DataFrame.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="pandas.DataFrame.dtypes.html#pandas.DataFrame.dtypes" title="pandas.DataFrame.dtypes"><code class="xref py py-obj docutils literal notranslate"><span class="pre">dtypes</span></code></a></p></td>
<td><p>Return the dtypes in the DataFrame.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="pandas.DataFrame.empty.html#pandas.DataFrame.empty" title="pandas.DataFrame.empty"><code class="xref py py-obj docutils literal notranslate"><span class="pre">empty</span></code></a></p></td>
<td><p>Indicator whether DataFrame is empty.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="pandas.DataFrame.iat.html#pandas.DataFrame.iat" title="pandas.DataFrame.iat"><code class="xref py py-obj docutils literal notranslate"><span class="pre">iat</span></code></a></p></td>
<td><p>Access a single value for a row/column pair by integer position.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="pandas.DataFrame.iloc.html#pandas.DataFrame.iloc" title="pandas.DataFrame.iloc"><code class="xref py py-obj docutils literal notranslate"><span class="pre">iloc</span></code></a></p></td>
<td><p>Purely integer-location based indexing for selection by position.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="pandas.DataFrame.index.html#pandas.DataFrame.index" title="pandas.DataFrame.index"><code class="xref py py-obj docutils literal notranslate"><span class="pre">index</span></code></a></p></td>
<td><p>The index (row labels) of the DataFrame.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="pandas.DataFrame.loc.html#pandas.DataFrame.loc" title="pandas.DataFrame.loc"><code class="xref py py-obj docutils literal notranslate"><span class="pre">loc</span></code></a></p></td>
<td><p>Access a group of rows and columns by label(s) or a boolean array.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="pandas.DataFrame.ndim.html#pandas.DataFrame.ndim" title="pandas.DataFrame.ndim"><code class="xref py py-obj docutils literal notranslate"><span class="pre">ndim</span></code></a></p></td>
<td><p>Return an int representing the number of axes / array dimensions.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="pandas.DataFrame.shape.html#pandas.DataFrame.shape" title="pandas.DataFrame.shape"><code class="xref py py-obj docutils literal notranslate"><span class="pre">shape</span></code></a></p></td>
<td><p>Return a tuple representing the dimensionality of the DataFrame.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="pandas.DataFrame.size.html#pandas.DataFrame.size" title="pandas.DataFrame.size"><code class="xref py py-obj docutils literal notranslate"><span class="pre">size</span></code></a></p></td>
<td><p>Return an int representing the number of elements in this object.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="pandas.DataFrame.style.html#pandas.DataFrame.style" title="pandas.DataFrame.style"><code class="xref py py-obj docutils literal notranslate"><span class="pre">style</span></code></a></p></td>
<td><p>Returns a Styler object.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="pandas.DataFrame.values.html#pandas.DataFrame.values" title="pandas.DataFrame.values"><code class="xref py py-obj docutils literal notranslate"><span class="pre">values</span></code></a></p></td>
<td><p>Return a Numpy representation of the DataFrame.</p></td>
</tr>
</tbody>
</table>

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html

In [8]:
holidays.columns

Index(['name', 'start_year', 'moon'], dtype='object')

In [9]:
holidays.index

Index(['3월', '5월', '음력4월', '8월', '10월', '12월'], dtype='object')

In [10]:
holidays.iloc[:2, :]

name  start_year   moon
3월   삼일절        1949  False
5월  어린이날        1975  False

In [11]:
holidays.loc[['음력4월', '12월'],['name', 'moon']]

name   moon
음력4월  부처님 오신 날   True
12월      기독탄신일  False

In [12]:
holidays['start_year'] > 1970

3월      False
5월       True
음력4월     True
8월      False
10월     False
12월     False
Name: start_year, dtype: bool

In [13]:
holidays[holidays['start_year'] > 1970]

name  start_year   moon
5월        어린이날        1975  False
음력4월  부처님 오신 날        1975   True

In [14]:
holidays.sort_values(by = ['start_year', 'name'], ascending = False)

name  start_year   moon
5월        어린이날        1975  False
음력4월  부처님 오신 날        1975   True
10월        한글날        1949  False
3월         삼일절        1949  False
12월      기독탄신일        1949  False
8월         광복절        1949  False

# 1. 데이터 파일 읽고 파악하기

기존에 있던 CSV(comma-separated values) 파일을 읽어와 DataFrame으로 저장하는 read_csv 함수를 알아봅니다. CSV는 각각의 요소를 쉼표(,)로 구분한 텍스트 파일로 포맷이 단순해 데이터를 저장하는 데 많이 쓰입니다.

In [1]:
help(pd.read_csv)

파일의 경로를 비롯해서 여러 매개변수가 있지만, 구분자(sep), 인코딩(encoding), csv파일의 첫 줄 처리(header) 정도를 자주 사용하게 됩니다. sep은 comma-seperated(,)가 아닌 tab(\t)등 쉼표가 아닌 다른 구분자를 사용하는 경우에, encoding은 저장 방식이 다를 때(윈도우의 기본은 cp949, 웹은 보통 utf-8 등), header는 파일의 첫 줄이 열의 이름이 아니라 바로 데이터로 시작할 때 바꾸어 줄 필요가 있습니다.

아래처럼 URL로 가져오는 것이 가장 편해 보이겠지만, 준비된 상태로 업로드된 데이터는 많지 않습니다. 다양한 경우에 대해서는 수업을 진행하면서 계속 다룰 예정입니다.

In [16]:
url = "https://raw.githubusercontent.com/eunyu1031/Media_interface_2020Spr/master/data/p19_mediapanel.csv"

data = pd.read_csv(url, encoding = 'utf-8')
# data = pd.read_csv(url, encoding = 'utf-8', header = None) # 열 이름 없이 바로 데이터로 시작하는 경우
data

pid      hid  area  area_siz  hhld_siz  fly_typ  gender  byear  \
0          10001      100     1         1         3        3       1   1967   
1          10002      100     1         1         3        3       2   1972   
2          20001      200     1         1         3        3       1   1965   
3          30001      300     8         1         3        3       1   1971   
4          30002      300     8         1         3        3       2   1974   
...          ...      ...   ...       ...       ...      ...     ...    ...   
10859  118290002  1182900    16         1         3        3       1   1983   
10860  118300001  1183000    16         1         2        3       1   1995   
10861  118310001  1183100    16         1         1        1       2   1939   
10862  118320001  1183200    16         1         1        1       2   1940   
10863  118330001  1183300    16         1         2        3       2   1947   

       age_gr  age  ...  news_med1  news_med2  news_med3 news_plat1  \
0           6   52  ...          3          4                     3   
1           5   47  ...          3          4                     3   
2           6   54  ...          4                                3   
3           5   48  ...          3                                3   
4           5   45  ...                                               
...       ...  ...  ...        ...        ...        ...        ...   
10859       4   36  ...                                               
10860       3   24  ...                                               
10861       8   80  ...                                               
10862       8   79  ...                                               
10863       8   72  ...                                               

      news_plat2  news_plat3 channel shop shop_dev shop_pay  
0              2                   1    1        1        2  
1              2                   3    1        1        2  
2                                  7                         
3                                  1                         
4                                  1    1        3        1  
...          ...         ...     ...  ...      ...      ...  
10859                              1                         
10860                              3    1        1        1  
10861                              6                         
10862                              7                         
10863                              7                         

[10864 rows x 66 columns]

데이터가 클수록 전체를 살펴보는 것은 효율적이지 않습니다. 다방면으로 간단하게 파악해 봅시다. 

__DataFrame.shape__는 행과 열의 수를 tuple로 return합니다.

In [17]:
data.shape

(10864, 66)

__DataFrame.head()__는 해당 DataFrame의 첫 5행을 보여줍니다. 괄호 안에 숫자를 넣으면 해당 숫자만큼 행을 보여줍니다.

In [18]:
data.head()

pid  hid  area  area_siz  hhld_siz  fly_typ  gender  byear  age_gr  age  \
0  10001  100     1         1         3        3       1   1967       6   52   
1  10002  100     1         1         3        3       2   1972       5   47   
2  20001  200     1         1         3        3       1   1965       6   54   
3  30001  300     8         1         3        3       1   1971       5   48   
4  30002  300     8         1         3        3       2   1974       5   45   

   ...  news_med1  news_med2  news_med3 news_plat1 news_plat2  news_plat3  \
0  ...          3          4                     3          2               
1  ...          3          4                     3          2               
2  ...          4                                3                          
3  ...          3                                3                          
4  ...                                                                      

  channel shop shop_dev shop_pay  
0       1    1        1        2  
1       3    1        1        2  
2       7                         
3       1                         
4       1    1        3        1  

[5 rows x 66 columns]

In [19]:
data.head(7)

pid  hid  area  area_siz  hhld_siz  fly_typ  gender  byear  age_gr  age  \
0  10001  100     1         1         3        3       1   1967       6   52   
1  10002  100     1         1         3        3       2   1972       5   47   
2  20001  200     1         1         3        3       1   1965       6   54   
3  30001  300     8         1         3        3       1   1971       5   48   
4  30002  300     8         1         3        3       2   1974       5   45   
5  30003  300     8         1         3        3       1   2002       2   17   
6  30004  300     8         1         3        3       1   2008       2   11   

   ...  news_med1  news_med2  news_med3 news_plat1 news_plat2  news_plat3  \
0  ...          3          4                     3          2               
1  ...          3          4                     3          2               
2  ...          4                                3                          
3  ...          3                                3                          
4  ...                                                                      
5  ...                                                                      
6  ...                                                                      

  channel shop shop_dev shop_pay  
0       1    1        1        2  
1       3    1        1        2  
2       7                         
3       1                         
4       1    1        3        1  
5       8                         
6                                 

[7 rows x 66 columns]

DataFrame.tail()은 뒤에서부터 다섯 행을 보여줍니다.

In [20]:
data.tail()

pid      hid  area  area_siz  hhld_siz  fly_typ  gender  byear  \
10859  118290002  1182900    16         1         3        3       1   1983   
10860  118300001  1183000    16         1         2        3       1   1995   
10861  118310001  1183100    16         1         1        1       2   1939   
10862  118320001  1183200    16         1         1        1       2   1940   
10863  118330001  1183300    16         1         2        3       2   1947   

       age_gr  age  ...  news_med1  news_med2  news_med3 news_plat1  \
10859       4   36  ...                                               
10860       3   24  ...                                               
10861       8   80  ...                                               
10862       8   79  ...                                               
10863       8   72  ...                                               

      news_plat2  news_plat3 channel shop shop_dev shop_pay  
10859                              1                         
10860                              3    1        1        1  
10861                              6                         
10862                              7                         
10863                              7                         

[5 rows x 66 columns]

여러 정보들을 계속 살펴봅시다.

In [21]:
data.columns

Index(['pid', 'hid', 'area', 'area_siz', 'hhld_siz', 'fly_typ', 'gender',
       'byear', 'age_gr', 'age', 'marry', 'income_gr', 'job', 'job_type',
       'job_etc', 'mob_use', 'mob_num', 'mob_csp', 'mob_dev', 'c01', 'c01_gr',
       'c02', 'c02_gr', 'c03', 'c03_gr', 'c04', 'c04_gr', 'c05', 'c05_gr',
       'app_pay', 'appin_pay', 'app_num', 'app_num_gr', 'appin_num',
       'appin_gr', 'c06', 'c06_gr', 'freq_app1', 'freq_app2', 'freq_app3',
       'sns_use', 'freq_sns1', 'freq_sns2', 'freq_sns3', 'cmt_board',
       'wt_board', 'wt_news', 'rating', 'wt_wiki', 'wt_sns', 'rating_sns',
       'sns_read', 'sns_write', 'sns_share', 'sns_eng', 'sns_priv',
       'news_med1', 'news_med2', 'news_med3', 'news_plat1', 'news_plat2',
       'news_plat3', 'channel', 'shop', 'shop_dev', 'shop_pay'],
      dtype='object')

__DataFrame.nunique()__는 중복되는 값 없이 해당 열(column)에 있는 유일한 값의 개수를 출력합니다.

In [22]:
data.nunique()

pid           10864
hid            4583
area             17
area_siz          2
hhld_siz          3
              ...  
news_plat3        5
channel           9
shop              3
shop_dev          4
shop_pay          7
Length: 66, dtype: int64

In [23]:
data['marry'].nunique()

4

__Series.unique()__는 중복되는 값 없이 해당 열(column)에 있는 유일한 값을 출력합니다.

In [24]:
data['marry'].unique()

array([2, 1, 3, 4], dtype=int64)

__Series.value_counts()__는 중복되는 값 없이 해당 열(column)에 있는 유일한 값이 각각 몇 번 나타나는지 빈도를 출력합니다. 

In [25]:
help(Series.value_counts)

Help on function value_counts in module pandas.core.base:

value_counts(self, normalize=False, sort=True, ascending=False, bins=None, dropna=True)
    Return a Series containing counts of unique values.
    
    The resulting object will be in descending order so that the
    first element is the most frequently-occurring element.
    Excludes NA values by default.
    
    Parameters
    ----------
    normalize : bool, default False
        If True then the object returned will contain the relative
        frequencies of the unique values.
    sort : bool, default True
        Sort by frequencies.
    ascending : bool, default False
        Sort in ascending order.
    bins : int, optional
        Rather than count values, group them into half-open bins,
        a convenience for ``pd.cut``, only works with numeric data.
    dropna : bool, default True
        Don't include counts of NaN.
    
    Returns
    -------
    Series
    
    See Also
    --------
    Series.count: Number 

In [26]:
data['marry'].value_counts()

2    6481
1    3286
3     858
4     239
Name: marry, dtype: int64

백분율을 보고 싶으면 normalize = True로 바꿉니다.

In [27]:
data['marry'].value_counts(normalize = True)

2    0.596557
1    0.302467
3    0.078976
4    0.021999
Name: marry, dtype: float64

value_counts()의 결과에서 기본적으로 가장 자주 등장하는 값이 먼저 정렬됩니다. 빈도가 아닌 인덱스(index) 순서대로 보고 싶으면 sort_values()가 아닌 sort_index()를 활용해야 합니다. 

In [28]:
data['marry'].value_counts().sort_index()

1    3286
2    6481
3     858
4     239
Name: marry, dtype: int64

__DataFrame.info()__는 DataFrame의 기본정보를 보여줍니다.
- DataFrame 'data'는 10864행 66열로 구성되었습니다.
- int 자료형인 열이 22개, object 자료형이 44개 있습니다. DataFrame에서는 문자열을 비롯해서 길이가 정해져 있지 않은 값은 object로 처리합니다.
- null은 비어있다는 표시입니다. 모든 데이터에 빈 값이 없습니다. 정말일까요?

In [29]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10864 entries, 0 to 10863
Data columns (total 66 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   pid         10864 non-null  int64 
 1   hid         10864 non-null  int64 
 2   area        10864 non-null  int64 
 3   area_siz    10864 non-null  int64 
 4   hhld_siz    10864 non-null  int64 
 5   fly_typ     10864 non-null  int64 
 6   gender      10864 non-null  int64 
 7   byear       10864 non-null  int64 
 8   age_gr      10864 non-null  int64 
 9   age         10864 non-null  int64 
 10  marry       10864 non-null  int64 
 11  income_gr   10864 non-null  int64 
 12  job         10864 non-null  int64 
 13  job_type    10864 non-null  object
 14  job_etc     10864 non-null  object
 15  mob_use     10864 non-null  int64 
 16  mob_num     10864 non-null  object
 17  mob_csp     10864 non-null  object
 18  mob_dev     10864 non-null  object
 19  c01         10864 non-null  object
 20  c01_gr

데이터를 가장 간단하게 요약해 보는 방법 중 하나는 기술통계(descriptive statistics)일 것입니다. __DataFrame.describe()__로 숫자 열(int, float)의 기술통계를 살펴볼 수 있습니다.

In [30]:
data.describe()

pid           hid          area      area_siz      hhld_siz  \
count  1.086400e+04  1.086400e+04  10864.000000  10864.000000  10864.000000   
mean   5.003951e+07  5.003950e+05      7.299521      1.094532      2.655284   
std    3.767155e+07  3.767155e+05      4.592933      0.292581      0.598260   
min    1.000100e+04  1.000000e+02      1.000000      1.000000      1.000000   
25%    1.640250e+07  1.640250e+05      3.000000      1.000000      2.000000   
50%    5.170000e+07  5.170000e+05      8.000000      1.000000      3.000000   
75%    6.890000e+07  6.890000e+05     11.000000      1.000000      3.000000   
max    1.183300e+08  1.183300e+06     17.000000      2.000000      3.000000   

            fly_typ        gender         byear        age_gr           age  \
count  10864.000000  10864.000000  10864.000000  10864.000000  10864.000000   
mean       2.835144      1.541053   1971.873711      5.196705     47.126289   
std        0.780984      0.498335     20.679911      1.985370     20.679911   
min        1.000000      1.000000   1915.000000      1.000000      6.000000   
25%        3.000000      1.000000   1957.000000      4.000000     30.000000   
50%        3.000000      2.000000   1970.000000      5.000000     49.000000   
75%        3.000000      2.000000   1989.000000      7.000000     62.000000   
max        5.000000      2.000000   2013.000000      8.000000    104.000000   

       ...           job       mob_use       sns_use    cmt_board  \
count  ...  10864.000000  10864.000000  10864.000000  10864.00000   
mean   ...      1.465022      1.037371      1.555504      1.26123   
std    ...      0.498798      0.189678      0.496933      0.81957   
min    ...      1.000000      1.000000      1.000000      1.00000   
25%    ...      1.000000      1.000000      1.000000      1.00000   
50%    ...      1.000000      1.000000      2.000000      1.00000   
75%    ...      2.000000      1.000000      2.000000      1.00000   
max    ...      2.000000      2.000000      2.000000      6.00000   

           wt_board       wt_news        rating       wt_wiki        wt_sns  \
count  10864.000000  10864.000000  10864.000000  10864.000000  10864.000000   
mean       1.176915      1.110549      1.136138      1.067102      1.631904   
std        0.648891      0.506810      0.537951      0.378722      1.258717   
min        1.000000      1.000000      1.000000      1.000000      1.000000   
25%        1.000000      1.000000      1.000000      1.000000      1.000000   
50%        1.000000      1.000000      1.000000      1.000000      1.000000   
75%        1.000000      1.000000      1.000000      1.000000      1.000000   
max        6.000000      6.000000      6.000000      6.000000      6.000000   

         rating_sns  
count  10864.000000  
mean       1.656020  
std        1.320204  
min        1.000000  
25%        1.000000  
50%        1.000000  
75%        1.000000  
max        6.000000  

[8 rows x 22 columns]

DataFrame.describe()의 결과도 하나의 DataFrame이므로 원하는 열의 값만 확인할 수도 있습니다.

In [31]:
data.describe()['age']

count    10864.000000
mean        47.126289
std         20.679911
min          6.000000
25%         30.000000
50%         49.000000
75%         62.000000
max        104.000000
Name: age, dtype: float64

# 2. 한국미디어패널의 인구통계학적 정보 살펴보기

먼저 한국 미디어패널의 인구통계학적 정보만 살펴보고자 합니다. 원래 파일에서 일부만 가져옵니다.

In [32]:
demo = data.iloc[:, :15]
demo.head()

pid  hid  area  area_siz  hhld_siz  fly_typ  gender  byear  age_gr  age  \
0  10001  100     1         1         3        3       1   1967       6   52   
1  10002  100     1         1         3        3       2   1972       5   47   
2  20001  200     1         1         3        3       1   1965       6   54   
3  30001  300     8         1         3        3       1   1971       5   48   
4  30002  300     8         1         3        3       2   1974       5   45   

   marry  income_gr  job job_type job_etc  
0      2         11    1        2          
1      2          1    2                2  
2      2         11    1        1          
3      2          9    1        1          
4      2          1    2                2

DataFrame 'demo'에서 각 열(column)의 의미는 아래와 같습니다.

<table>
<thead>
<tr>
<td width="91">
<p>Original</p>
</td>
<td width="97">
<p>Cleaned</p>
</td>
<td width="210">
<p>Description</p>
</td>
<td>
<p>Detail</p>
</td>
</tr>
</thead>
<tbody>
<tr>
<td width="91">
<p>pid</p>
</td>
<td width="97">
<p><strong>pid</strong></p>
</td>
<td width="210">
<p>[개인정보] 개인 통합 ID</p>
</td>
<td>&nbsp;</td>
</tr>
<tr>
<td width="91">
<p>hid</p>
</td>
<td width="97">
<p><strong>hid</strong></p>
</td>
<td width="210">
<p>[개인정보] 가구 통합 ID</p>
</td>
<td>&nbsp;</td>
</tr>
<tr>
<td width="91">
<p>p__area</p>
</td>
<td width="97">
<p><strong>area</strong></p>
</td>
<td width="210">
<p>[개인정보] 지역</p>
</td>
<td>
<p>1) 서울 2) 부산 3) 대구 4) 인천 5) 광주 6) 대전 <br>
    7) 울산 8) 경기 9) 강원 10) 충북 11) 충남 12) 전북 <br>
    13) 전남 14) 경북 15) 경남 16) 제주 17) 세종</p>
</td>
</tr>
<tr>
<td width="91">
<p>p__area_siz</p>
</td>
<td width="97">
<p><strong>area_siz</strong></p>
</td>
<td width="210">
<p>[가구정보] 지역구분</p>
</td>
<td>
<p>① 시부 ② 군부</p>
</td>
</tr>
<tr>
<td width="91">
<p>p__hhldsiz</p>
</td>
<td width="97">
<p><strong>hhld_siz</strong></p>
</td>
<td width="210">
<p>[가구정보] 가구원 수</p>
</td>
<td>
<p>① 1인 가구 ② 2인 가구 ③ 3인 이상 가구</p>
</td>
</tr>
<tr>
<td width="91">
<p>p__fly_typ</p>
</td>
<td width="97">
<p><strong>fly_typ</strong></p>
</td>
<td width="210">
<p>[가구정보] 가족 구성</p>
</td>
<td>
<p>1) 1인가구 2) 1세대가구(부부) 3) 2세대가구(부부+자녀) <br>
    4) 3세대가구 (조부모+부부+자녀) 5) 기타</p>
</td>
</tr>
<tr>
<td width="91">
<p>p__gender</p>
</td>
<td width="97">
<p><strong>gender</strong></p>
</td>
<td width="210">
<p>[개인정보] 성별</p>
</td>
<td>
<p>1)남 2)여</p>
</td>
</tr>
<tr>
<td width="91">
<p>p__byear</p>
</td>
<td width="97">
<p><strong>byear</strong></p>
</td>
<td width="210">
<p>[개인정보] 출생연도</p>
</td>
<td>&nbsp;</td>
</tr>
<tr>
<td width="91">
<p>p__age</p>
</td>
<td width="97">
<p><strong>age_gr</strong></p>
</td>
<td width="210">
<p>[개인정보] 나이(리코드)</p>
</td>
<td>
<p>1) 만 10대 미만 2) 만 10-19세 3) 만 20-29세 4) 만 30-39세 <br>
    5) 만 40-49세 6) 만 50-59세 7) 만 60-69세 8) 만 70세 이상</p>
</td>
</tr>
<tr>
<td width="91">
<p>p__age1</p>
</td>
<td width="97">
<p><strong>age</strong></p>
</td>
<td width="210">
<p>[개인정보] 나이(만 연령)</p>
</td>
<td>&nbsp;</td>
</tr>
<tr>
<td width="91">
<p>p__mar</p>
</td>
<td width="97">
<p><strong>marry</strong></p>
</td>
<td width="210">
<p>[개인정보] 결혼 여부</p>
</td>
<td>
<p>① 미혼 ② 배우자 있음 ③ 사별 ④ 이혼</p>
</td>
</tr>
<tr>
<td width="91">
<p>p__income1</p>
</td>
<td width="97">
<p><strong>income_gr</strong></p>
</td>
<td width="210">
<p>[개인정보] 개인 월평균 소득</p>
</td>
<td>
<p>1) 소득 없음 2) 50만원 미만 3) 50-100만원 미만 4) 100-150만원 미만 <br>
    5) 150-200만원 미만 6) 200-250만원 미만 7) 250-300만원 미만 8) 300-350만원 미만  <br>
    9) 350-400만원 미만 10) 400-450만원 미만 11) 450-500만원 미만 12) 500-550만원 미만  <br>
    13) 550-600만원 미만 14) 600-650만원 미만 15) 650-700만원 미만 16) 700-750만원 미만  <br>
    17) 750-800만원 미만 18) 800만원 이상</p>
</td>
</tr>
<tr>
<td width="91">
<p>p__job1</p>
</td>
<td width="97">
<p><strong>job</strong></p>
</td>
<td width="210">
<p>[개인정보] 직업 유무</p>
</td>
<td>
<p>① 예 ② 아니오</p>
</td>
</tr>
<tr>
<td width="91">
<p>p__job2</p>
</td>
<td width="97">
<p><strong>job_type</strong></p>
</td>
<td width="210">
<p>[개인정보] 직업 종사상 지위</p>
</td>
<td>
<p>① 임금 근로자 ② 고용주(고용인이 있는 자영업자) <br>
    ③ 단독 자영업자(고용인이 없는 자영업자) ④ 무급 가족 종사자</p>
</td>
</tr>
<tr>
<td width="91">
<p>p__job4</p>
</td>
<td width="97">
<p><strong>job_etc</strong></p>
</td>
<td width="210">
<p>[개인정보] 무직 구분</p>
</td>
<td>
<p>① 학생 ② 전업주부 ③ 군인(직업군인 제외) ④ 기타/무직</p>
</td>
</tr>
</tbody>
</table>

<p style="color:#0000FF";><font size = "5"><b>
 # 연습하기
    </b></font></p>
    
#### DataFrame 'demo'에서 다음 정보들을 확인하세요.

- 행과 열의 수

- 첫 10행

- 기본정보

- 모든 열에 있는 유일한 값의 개수 (nunique)

- 남녀의 비율 (value_counts)

- 각 연령대별 사람 수 (오름차순으로 정렬)

- 각 지역별 사람 수 (오름차순으로 정렬)

- job_etc의 유일한 값 (unique)

- 기술통계